adagrams/game.py

```diff
@@ -1,11 +1,105 @@
+import random
+from collections import defaultdict
+
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
 def draw_letters():
-    pass
+    letter_list = []
+    letter_freq = defaultdict(int)
+
+    while len(letter_list) < 10:
+        letter = random.choice(list(LETTER_POOL.keys()))
+        if letter_freq[letter] < LETTER_POOL[letter]:
+            letter_list.append(letter)
+            if letter in letter_freq:
+                letter_freq[letter] += 1
+
+    return letter_list
 
 def uses_available_letters(word, letter_bank):
-    pass
+    is_anagram = True
+    copy_letter_bank = letter_bank[:]
+    word = word.upper()
+
+    for letter in word:
+        if letter not in copy_letter_bank:
```

Comment:
Notice here that you need to loop through every letter in word on line 51. Then you're using the `in` operator to check if the letter is in the copy_letter_bank on line 52. Recall that using the `in` operator has O(n) time complexity, as does your for loop, so we've got O(n^2).

Rather than using a list of the letters in the hand, could we build a helper data structure (like a dictionary) that could let us look up the number of tiles remaining of each type?

If you have a frequency dict and then check if a key is in the dict, then you can leverage the fact that dicts constant time lookup (in contrast to the list's linear time look up).